src/input/input_json.rs

```diff
@@ -34,63 +36,74 @@ impl<'a> Input<'a> for JsonInput {
     }
 
     fn is_none(&self) -> bool {
-        matches!(self, JsonInput::Null)
+        matches!(self, DataValue::Null)
     }
 
     fn as_kwargs(&'a self, py: Python<'a>) -> Option<&'a PyDict> {
         match self {
-            JsonInput::Object(object) => {
+            DataValue::Object(object) => {
                 let dict = PyDict::new(py);
                 for (k, v) in object.iter() {
                     dict.set_item(k, v.to_object(py)).unwrap();
                 }
                 Some(dict)
             }
+            DataValue::Py(obj) => obj.as_ref(unsafe { Python::assume_gil_acquired() }).as_kwargs(py),
```

Comment:
you have `py` here, so this shouldn't be necessary?

Reply:
Probably, we blindly added this everywhere. Needs cleanup.